### ISSUE TITLE
Exceptions not captured in Google Cloud Function 

### ISSUE THREAD

**Issue by LarsKlingen** [getsentry/sentry-python#1078]
Hello,
I am trying to capture erros happening in GCPs Google Cloud Functions with Sentry. I initialize sentry as shown in the documentation for cloud function, but exceptions never appear in the dashboard. Running the below function locally successfully sents the logs to Sentry. Also when I explicitly capture the error and use capture_exception the exception is sent to Sentry successfully from GCPs cloud function. 
I also tried moving the sentry_sdk.init inside the function, but that also didnt help. Im using version 1.0.0, but also tried with earlier versions. The cloud function is triggered with GCPs pub/sub. 
Thanks!

```
from google.cloud import bigquery
import redis
import numpy as np
import json
import os
import sentry_sdk
from sentry_sdk.integrations.gcp import GcpIntegration
from sentry_sdk import capture_exception

sentry_sdk.init(
    dsn="working-dsn-key",
    integrations=[GcpIntegration(timeout_warning=True)],
    traces_sample_rate=1.0,  # adjust the sample rate in production as needed
)


def update_table(response, metadata):
    # This will give an error, but it wont be logged, only running locally.
    response['blabla']

    # This will give an error and successfully sent to sentry from cloud function
    # try:
    #     response['blabla']
    # except Exception as e:
    #     # Alternatively the argument can be omitted
    #     capture_exception(e)

    return {'finished': 200}
```

**Comment by frankie567** [comment]
Same here, no exception captured on Pub/Sub triggered Cloud Functions.

* Cloud Functions Runtime: Python 3.8
* sentry_sdk: 1.1.0

**Comment by frankie567** [comment]
I've looked a bit deeper this problem, and I found two things:

1. The integration is only compatible with Python 3.7 runtime. There is a subtle warning, but I think this should be mentioned more clearly [in the documentation](https://docs.sentry.io/platforms/python/guides/gcp-functions/). If you use another runtime (like Python 3.8 or 3.9), it does nothing more than standard Sentry configuration.
2. When enabling Sentry debug mode, I clearly see errors being catched and added to the worker queue. However, my guess is that when an error is thrown, Google abruptly kills the function ; not giving chance for the queue to be processed before exit. Therefore, errors are not reported to Sentry.

I ended up writing a decorator that manually catches exceptions, reports them to Sentry and forces the queue to flush. This way, the errors are reported:

```py
def sentry(func):
    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        try:
            func(*args, **kwargs)
        except Exception as e:
            sentry_sdk.capture_exception(e)
            sentry_sdk.flush()
            raise e

    return wrapper

@sentry
def pubsub_cloud_function(event, context):
    raise Exception("Error!")
```
